lib/hash.c

```diff
@@ -141,26 +141,24 @@ void *hash_get(struct hash *hash, void *data, void *(*alloc_func)(void *))
 	unsigned int key;
 	unsigned int index;
 	void *newdata;
-	unsigned int len;
 	struct hash_backet *backet;
 
 	key = (*hash->hash_key)(data);
 	index = key & (hash->size - 1);
-	len = 0;
 
 	for (backet = hash->index[index]; backet != NULL;
 	     backet = backet->next) {
 		if (backet->key == key && (*hash->hash_cmp)(backet->data, data))
 			return backet->data;
-		++len;
 	}
 
 	if (alloc_func) {
 		newdata = (*alloc_func)(data);
 		if (newdata == NULL)
 			return NULL;
 
-		if (len > HASH_THRESHOLD) {
+		double lf = (hash->count + 1) / (double) hash->size;
```

Comment:
please, no FP division here...

```
/* 100% threshold */
#define HASH_THRESHOLD(used, size) ((used) > (size))
/* 80% threshold */
#define HASH_THRESHOLD(used, size) ((used) * 5U > (size) * 4U)
/* ... */

if (HASH_THRESHOLD(hash->count + 1, hash->size))
```

(if this integer-overflows, we're OOM anyway)